### PR TITLE
Publish javaagent to Sonatype

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -15,6 +15,13 @@ build:
   script:
     - ./gradlew build -x :smoke-tests:test -x :testing:profiler-tests:test --scan --no-daemon --stacktrace
 
+snapshot:
+  stage: release
+  rules:
+    - if: '$CI_COMMIT_REF_NAME == "main"'
+  script:
+    - ./gradlew snapshot --no-daemon --stacktrace
```
Comment (iNikem) [inline .gitlab-ci.yml:23]: This does not need repository closing and releasing and whatnot?

File: agent/build.gradle [hunk 2]
```diff
@@ -1,88 +1,150 @@
...
+  }
+
+  assemble {
+    dependsOn(shadowJar)
+  }
+}
+
+rootProject.tasks.named("release").configure {
+  finalizedBy(tasks["publishToSonatype"])
+}
```
Comment (iNikem) [inline agent/build.gradle:102]: It's groovy, does not `release.finalizedBy publishToSonatype` work?

File: agent/build.gradle [hunk 3]
```diff
@@ -1,88 +1,150 @@
...
+          license {
+            name = "The Apache License, Version 2.0"
+            url = "http://www.apache.org/licenses/LICENSE-2.0.txt"
+          }
+        }
+        developers {
+          developer {
+            id = "splunk"
+            name = "Splunk Instrumentation Authors"
+            email = "support+sonatype@signalfx.com"
```
Comment (iNikem) [inline agent/build.gradle:129]: Is this email correct?

File: build.gradle [hunk 4]
```diff
@@ -4,13 +4,29 @@ plugins {
...
+
+  repositories {
+    sonatype {
+      username = System.getenv('SONATYPE_USERNAME')
+      password = System.getenv('SONATYPE_PASSWORD')
+    }
+  }
+}
+
+group = "com.splunk"
```
Comment (iNikem) [inline build.gradle:29]: For some reason this looks like duplication with changes in `agent/build.gradle`. Why do we need both?

File: .gitlab-ci.yml (hunk 1 above)
Comment (mateuszrzeszutek) [reply .gitlab-ci.yml:23]: I think it doesn't: staging, closing and releasing are just for non-snapshot versions.

File: agent/build.gradle (hunk 2 above)
Comment (mateuszrzeszutek) [reply agent/build.gradle:102]: Yeah, it'd work, but I don't want to add more work for the Kotlin migration (coming soon)

File: build.gradle (hunk 4 above)
Comment (mateuszrzeszutek) [reply build.gradle:29]: Hmm, the project group seems to be empty for `:agent`, no idea why. I've decided to just duplicate that for the time being, before I figure out why.

Comment (iNikem) [reply build.gradle:29]: Sorry, I got confused actually :) Over there is configuring publishing as in "what jars and pom" and signing, over here is configuring publishing as in "where to push". My mistake, sorry.

**Review by iNikem — APPROVED** [review]

**Review by breedx-splk — APPROVED** [review]
Those reformatting changes were also very exciting.  🙃

**Pull request merged**